ci/test-sanity.sh

```diff
@@ -53,4 +53,22 @@ _ ci/check-channel-version.sh
 _ ci/nits.sh
 _ ci/check-ssh-keys.sh
 
+scripts/increment-cargo-version.sh check
+
+if ! [ -v SOLANA_CI_ALLOW_STALE_CARGO_LOCK ] ; then
+# Disallow uncommitted Cargo.lock changes
+(
+  _ scripts/cargo-for-all-lock-files.sh tree >/dev/null
+  set +e
+  if ! _ git diff --exit-code; then
```

Comment:
theoretically this should be `git diff --exit-code :*Cargo.lock` to make the comment accurate. might do so in a followup